src/main/java/io/jenkins/docker/pipeline/DockerNodeStepExecution.java

```diff
@@ -193,7 +193,7 @@ private void invokeBody(DockerTransientNode node, TaskListener listener) {
             env.overrideExpandingAll(computer.buildEnvironment(listener));
             env.put("NODE_NAME", computer.getName());
             env.put("EXECUTOR_NUMBER", "0");
-            env.put("NODE_LABELS", Util.join(node.getAssignedLabels(), " "));
+            env.put("NODE_LABELS", node.getAssignedLabels().stream().map(Object::toString).collect(Collectors.joining(" ")));
```

Comment:
While I'm unsure of the cause, it looks like this change breaks the unit-tests.

...also, I'm not a big fan of having so many method calls daisy-chained on the same line - it makes debugging a NPE nigh-on impossible.
I'd suggest implementing a `join` method (possibly in this plugin's utils package, maybe as a new class unless it fits well in another), and adding a unit-test for it (to prove it'll cope with a null input, or an input containing nulls, as well as normal non-null stuff), and then call that instead of `Util.join`.
I _suspect_ that such a unit-test would then reveal that `Util.join` was more null-safe than this new code and fixing that would then fix the failing unit-tests.